src/Fetcher/fetcher.go

```diff
@@ -86,61 +124,72 @@ func errHandler(err error, params ...string) {
 	}
 }
 
-func executer(cmd *exec.Cmd) error {
+func executer(cmd *exec.Cmd, folder string) error {
 	//TODO: Figure out a way to take strings as input and build cmd
+	cmd.Dir = folder
 	_, err := cmd.CombinedOutput()
 
 	return err
 }
 
 func puller(url, branch, specFile string) ([]string, error) {
 	paths := []string{}
+	
+	// Create temp folder for git in the system temp folder
+	var randomNum int
+	var tempFolder string
+	for {
+		randomNum = rand.Int()
+		tempFolder = fmt.Sprintf("temp/%d", randomNum)
+		if _, err := os.Stat(tempFolder); os.IsNotExist(err) {
+			os.Mkdir(tempFolder, 0700)
+			break
+		}
+	}
 
 	// Create dummy repo
 	cmd := exec.Command("git", "init")
-	err := executer(cmd)
+	err := executer(cmd, tempFolder)
 	if err != nil {
 		return paths, err
 	}
 
 	//Enable sparse Checkout
 	cmd = exec.Command("git", "config", "core.sparseCheckout", "true")
-	err = executer(cmd)
+	err = executer(cmd, tempFolder)
 	if err != nil {
 		return paths, err
 	}
 
 	// Add whitelist to sparse-checkout
 	fileData, err := os.ReadFile(specFile)
-	err = os.WriteFile(".git/info/sparse-checkout", fileData, 0644)
 	if err != nil {
 		return paths, err
 	}
 
-	// Add remote repo
-	cmd = exec.Command("git", "remote", "add", "origin", url)
-	err = executer(cmd)
+	err = os.WriteFile(tempFolder + "/.git/info/sparse-checkout", fileData, 0644)
 	if err != nil {
 		return paths, err
 	}
 
-	// git pull from remote repo
-	cmd = exec.Command("git", "pull", "origin", branch, "--depth=1")
-	err = executer(cmd)
+	// Add remote repo
+	cmd = exec.Command("git", "remote", "add", "origin", url)
+	err = executer(cmd, tempFolder)
 	if err != nil {
 		return paths, err
 	}
 
-	//remove .git folder
-	err = os.RemoveAll("./.git")
+	// git pull from remote repo
+	cmd = exec.Command("git", "pull", "origin", branch, "--depth=1")
+	err = executer(cmd, tempFolder)
 	if err != nil {
 		return paths, err
 	}
```

Comment:
Oh, how come we don't need to remove the .git folder anymore? O:

Reply:
We put all the repo folders into a temp folder which is deleted when everything is fetched, then removing all the repo folders